Fix direction of library implication for --allow-warnings

keyboardDrummer opened the pull request:
### Description
Fix the direction in which a consumer enforces --allow-warning to be used in a library

### How has this been tested?
Manually

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

robin-aws reviewed (changes requested):
I think this needs an explicit test to confirm, especially since it's the only case where an "incompatible" doo file produces a warning instead of an error.

What was the root cause of the NPE?

keyboardDrummer commented:
> I think this needs an explicit test to confirm, especially since it's the only case where an "incompatible" doo file produces a warning instead of an error.

Added

> What was the root cause of the NPE?

Err, somehow the fix wasn't in the PR yet. Added it now and a test for it.
